src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,112 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status    - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    try {
+      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+
+      Map<String, List<Long>> failedUpdates = new HashMap<>();
+
+      for (Long userId : userIds) {
+        modifyTeacherPendingStatusForUser(userId, requestingUser, status, failedUpdates);
+      }
+
+      if (!failedUpdates.isEmpty()) {
+        String errorMessage = "";
+
+        if (failedUpdates.containsKey(USERS_NOT_FOUND)) {
+          errorMessage = String.format("One or more users could not be found: %s. ",
+              failedUpdates.get(USERS_NOT_FOUND));
+        }
+
+        if (failedUpdates.containsKey(FAILED_TO_SEND)) {
+          errorMessage += String.format("Emails could not be sent to userIds: %s",
+              failedUpdates.get(FAILED_TO_SEND));
+        }
+        return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
+      }
+
+    } catch (NoUserLoggedInException e) {
+      return SegueErrorResponse.getNotLoggedInResponse();
+    } catch (SegueDatabaseException e) {
+      log.error("Database error while trying to change teacher_pending status", e);
+      return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR,
+          "Could not save new teacher_pending status to the database").toResponse();
+    }
+
+    return Response.ok().build();
+  }
+
+  private void modifyTeacherPendingStatusForUser(Long userId, RegisteredUserDTO requestingUser, Boolean status,
+                                                 Map<String, List<Long>> failedUpdates) throws SegueDatabaseException {
+
+    try {
+      RegisteredUserDTO user = this.userManager.getUserDTOById(userId);
+
+      if (null == user) {
+        throw new NoUserException("No user found with this ID.");
+      }
+
+      Boolean oldStatus = user.getTeacherPending();
+      this.userManager.updateTeacherPendingFlag(userId, status);
+      if (log.isInfoEnabled()) {
```

Comment:
I don't think this is needed. You can simply call `log.info`, and if it's not enabled then it would be a no-op.

Reply:
I believe this was a change sonarlint wanted - I'll change it back and see if it objects!

Reply:
I think it may have complained before if you were using string concatenation in the `log.info` string, but you don't seem to be doing that now, so it should be fine.